src/omezarr-helper.ts

```diff
@@ -432,9 +432,11 @@ function generateNeuroglancerStateForOmeZarr(
   return encodeURIComponent(stateJson);
 }
 
-async function getZarrArray(dataUrl: string): Promise<zarr.Array<any>> {
+async function getZarrArray(
+  dataUrl: string,
+  zarrVersion: 2 | 3
+): Promise<zarr.Array<any>> {
   log.debug('Getting Zarr array for', dataUrl);
-  const zarrVersion = 2;
   const store = new zarr.FetchStore(dataUrl);
   return await omezarr.getArray(store, '/', zarrVersion);
```

Comment:
This pre-exists this pull request, but a dimension separator of `.` is technically the default Zarr v2 separator. We have most often seen this from Zarr arrays sent to us from outside contractors such as Ariadne. I wonder if too many low-level Zarr details are passing through the abstraction barrier here.

Perhaps we should just detect the presence of a `.zarray`, `.zattrs`, `.zgroup`, or `zarr.json` and then let ome-xml.js or zarrita.js "open" the array, parse (and maybe validate the JSON), and then give us details such as zarr version.